src/routes/startGame.ts

```diff
@@ -39,58 +39,43 @@ router.get(
       return res.sendStatus(401)
     }
 
-    const client = getClient()
-    client
-      .connect()
-      .then(async () => {
-        const { gameId, startingFunds } = await getGameInfoIfNotStarted(
-          creatorToken,
-          client
-        )
-        if (gameId === null) {
-          return res.sendStatus(400)
-        }
-
-        const players = await getPlayersInGame(gameId, client)
-
-        if (players.length < 2) {
-          return res.sendStatus(402)
-        }
-
-        const playersInGame = convertToInternalPlayerInfo(players)
-
-        const cardDeck = fullCardDeck.slice()
-
-        const gameInfo = createStartedGameInfo(playersInGame, cardDeck)
-
-        await updatePlayersStates(
-          playersInGame,
-          startingFunds,
-          gameInfo,
-          client
-        )
-
-        const smallBlind = await getSmallBlind(gameId, players.length, client)
-        await updateGameState(
-          playersInGame[0].token,
-          gameId,
-          gameInfo,
-          client,
-          smallBlind,
-          players.length
-        )
-
-        await notifyPlayers(playersInGame, gameInfo)
-
-        res.sendStatus(200)
-      })
-      .catch(async (err) => {
-        console.log(err.stack)
-        return res.sendStatus(500)
-      })
-      .finally(async () => {
-        await client.end()
-      })
+    await runRequestWithClient(res, async (client) => {
+      const { gameId, startingFunds } = await getGameInfoIfNotStarted(
+        creatorToken,
+        client
+      )
+      if (gameId === null) {
+        return res.sendStatus(400)
+      }
+
+      const players = await getPlayersInGame(gameId, client)
+
+      if (players.length < 2) {
+        return res.sendStatus(402)
+      }
+
+      const playersInGame = convertToInternalPlayerInfo(players)
+
+      const cardDeck = fullCardDeck.slice()
+
+      const gameInfo = createStartedGameInfo(playersInGame, cardDeck)
+
+      await updatePlayersStates(playersInGame, startingFunds, gameInfo, client)
+
+      const smallBlind = await getSmallBlind(gameId, players.length, client)
+      await updateGameState(
```

Comment:
I think we should do something about this function. If we use function such as `addPlayersToGameInfo` maybe we should extend the interface of startingGameInfo with `gameId: number, 
smallBlindWho: string` (I think they are properties of staring game), and send it here?  What do you think?

Reply:
This is a good point, but we will do this in a separate PR. I improved the `joinGame` request, but later decided to not improve the other endpoints. They all need improvement, but it's out of the scope of this PR